src/bcf/header.rs

```diff
@@ -240,6 +257,54 @@ impl HeaderView {
             (*(*self.inner).id[htslib::BCF_DT_SAMPLE as usize].offset(*id as isize)).key) };
         key.to_bytes().to_vec()
     }
+
+    /// Return structured `HeaderRecord`s.
+    pub fn header_records(&self) -> Vec<HeaderRecord> {
```

Comment:
I would return a HashMap here, which assigns keys to HeaderRecord.

Reply:
We can use LinearMap instead, there are only a few keys involved here. See https://docs.rs/linear-map

Reply:
Right. And moreover there might be duplicate keys I think. Returning a Vector is fine.

Reply:
Hm, so should I just rebase or perform any changes?

Reply:
OK, resolving the merge conflicts now FWIW, would have to be done anyway.